src/node_crypto.cc

```diff
@@ -5263,7 +5263,7 @@ void PBKDF2(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_IF_NOT_BUFFER(args[1], "Salt");
 
   pass = static_cast<char*>(node::Malloc(passlen));
-  if (pass == nullptr) {
+  if (pass == nullptr && passlen > 0) {
     FatalError("node::PBKDF2()", "Out of Memory");
   }
```

Comment:
The `if (pass == nullptr)` branch would never execute since you're already testing the "truthiness" of `pass` up above. Besides, I'm not sure that the "assignment and test" inside a conditional is really encouraged in node core (at least in C++).